frontend/src/ui/wallet/shared/confirm-intention-view.js

```diff
@@ -0,0 +1,33 @@
+import React, {useEffect, useState} from 'react'
+import {AccountAddress} from '@stellar-expert/ui-framework'
+import {fetchAssetPrices} from "../../../state/ledger-data/asset-price"
+
+export function ConfirmIntentionView({transfer}) {
+    const selfTransfer = transfer.source === transfer.destination
+    const [estimatedText, setEstimatedText] = useState('')
+    const accountName = window.predefinedAccountDisplayNames[transfer.source]
+
+    useEffect(() => {
+        (async () => {
+            try {
+                const prices = await fetchAssetPrices(transfer.network, [transfer.asset[0]])
+                const estimatedValue = Object.values(prices)[0]
+                const estimatedPrice = transfer.amount[0] * estimatedValue
+                if (estimatedPrice) setEstimatedText(`(~${estimatedPrice.toFixed(2)}$)`)
+            } catch (e) {
+                console.log(e)
+            }
+        })()
+    }, [transfer])
+
+    if (selfTransfer)
+        return <div className="double-space">
+            You are about to swap {transfer.amount[0]} {transfer.asset[0].split('-')[0]} {estimatedText} for&nbsp;
+            {transfer.amount[1]} {transfer.asset[1].split('-')[0]}
+        </div>
+    return <div className="double-space">
+        Transfer {transfer.amount[0]} {transfer.asset[0].split('-')[0]} {estimatedText} from [{accountName}] to&nbsp;
```

Comment:
Use `<AccountAddress/>` instead of `accountName`. It will retrieve the friendly account name automatically.